### DISCUSSION TITLE
How To Update rAudio-1

### DISCUSSION THREAD

**Discussion by teslasparks301** [rern/rAudio#479] [General]
Is it safe to run 'pacman -Syu'? Or should all updates be done strictly through the Add-Ons page of the web gui?

**Comment by klassikmann** [comment]
I'm updating my installation applying 'pacman -Syu' almost daily. No problems so far.

**Reply by teslasparks301** [reply]
I ran the pacman update and ended up with one problem. The Radio Paradise radio links don't work. The radio links included with the distro are the only radio links I have. Everything else seems to be OK. There was a question about installing Jack2 that I relied yes to. Don't know if that has anything to do with the problem.

**Reply by teslasparks301** [reply]
I setup a log file for mpd and the telling portion of the error was:
CURL failed: SSL certificate problem: certificate is not yet valid

Turns out my system date was about 10 days behind. I installed the ntp package and ran ntpdate. Started working just fine. Now I'll need to figure out my approach to maintaining system time.

**Comment by R101G** [comment]
@teslasparks301 
My system time has been OK since I entered `systemctl restart systemd-timesync` and `systemctl restart systemd-networkd.` I am not sure if both commands are needed and it may just be coincidence, or a temporary effect.

This bug does not appear to affect everyone, and might be because systemd does not see the network. Just out of interest, are you using Ethernet or wifi?

**Reply by teslasparks301** [reply]
I have never heard of systemd-timesyncd.  I checked them both and both were already enabled though timesync looks like it hung on Aug 8 which is the date my system time was showing before resetting it.  networkd just brings up the network which, of course, is needed to connect to a time server.  It's really just a supporting service to set time while timesyncd does the actual work.  I'm not sure why systemd needs to set time.  There already were tools for that.  It's really turned into Frankenstein's monster.

<geezer-talk>I started linux with kernel 1.1.13.  It's hard to keep up with the changes after learning how to manage the system.  And don't even get me started on the revamp of resolv.conf.</geezer-talk>

I installed fake-hwclock and ntp, changed /etc/ntp.conf to use my regional ntp pool, and enabled the two of services. I think I'll be disabling systemd-timesyncd. I have probably ensured future headaches for myself when I do a reinstall or update. hahaha

BTW, my network connection is wifi.  Since rAudio runs headless, I would have noticed not being able to connect to snapcast, ssh, or the web interface.

We've really drifted off-topic but I enjoy the conversation. :)

**Comment by R101G** [comment]
@teslasparks301 
As with all things Arch, it is very difficult to find an explanation of how systemd-timesyncd works, but our current problem seems to be that it is waiting for systemd-networkd to see an active network connection, even though there already is one.

As @klassikmann  does not have the problem, I suspect it is just wifi users who are affected. That would explain why the time was correct immediately after my fresh install made with a wire connection, but drifted once I went back to wifi. Unfortunately, there is no good way to poll users about their setups. I would not mind at all if @rern incorporated this option in the OS.

Your report has linked both of my current bug reports, one for wrong time and the other the Radio Paradise problem. It did not occur to me that the time being out of sync would affect the updates.

**Reply by teslasparks301** [reply]
I haven't dug very deep into Paradise problem but from what I can tell curl would not handshake with the security certificate because the system time and actual time differential was too great. So I don't think there is an actual problem with Paradise but just system time.  I ran a pacman -Syu and rebooted the box.  When I ssh connected to the rpi again I systime was about 2 minutes slow. 3-5 minutes later it was within a couple of milliseconds.  

To summarize, I have ntpd, fake-hwclock, and systemd-timesyncd.service all working and so far seems to be OK.

At no time did I use anything but wifi.

**Comment by R101G** [comment]
From what I read, the time and date problem was caused by a systemd update. Hopefully it will be fixed by another update. So far, I find that entering `systemctl restart systemd-networkd` after every reboot is a workaround.

**Reply by teslasparks301** [reply]
After a quick look at systemd-timesyncd.service, systemd-networkd.service, and a couple of other service files plus the man page timesyncd my guess is that timesyncd is trying to run before networkd is done setting up your wifi.  I found no dependencies between the 2.   CAUTION!! Rant Ahead!  Just because I didn't see them doesn't mean there not there.  systemd is a huge steaming pile of Borg-ish code with it's little fingers invading so much of the administrative code for no good reason that I can understand.  fake-hwclock, ntpdate, and ntpd were already well established and can do a better job.  The timesyncd man page hints at that.  But that's just me.  In the interest of brevity, I won't tell you how I really feel.  ;) Rant Off!

The short story is try making sure wifi is established before timesyncd runs and see if that works. I would add a dependency to systemd.timesyncd.service.  That's essentially what you are doing with the restarts. And/Or, just for fun, try a static address instead of DHCP on your wifi.  The interface up time will be faster using a static address.

**Comment by R101G** [comment]
@teslasparks301 It looks to me as if the files in /etc/systemd/network/ tell systemd-networkd what interfaces are valid, and the wifi one is missing. Just having one file with Name=* in it is probably all that is needed, but someone decided to complicate things and forgot to include wifi interfaces.

Try my possible fix in #472 and see if it works for you too. It would be good to have confirmation or otherwise, and there doesn't seem to be many of us using wifi.

**Reply by teslasparks301** [reply]
I shutdown my box and let it sit for about an hour. When I booted my system time was about 1 hour slow. That was to be expected since it was set via fake-hwclock. In about 5 min ntpd had me within several milliseconds of true time. I checked the status of timesyncd and found it still wasn't running. I applied the fix suggested in [#472](https://github.com/rern/rAudio-1/discussions/472) and redid the shutdown, wait, and boot. Time was within several milliseconds as soon as it booted because timesyncd (which could be done by ntpdate) was running. The fix worked for timesyncd. I decided to disable timesyncd and go with fake-hwclock and ntpd. I don't know if that will cause a problem after a prolonged shutdown, say a few weeks or more, but that's the route I'm going for now. 

There is more than one way to do it (TIMTOWTDI - the motto of Perl - pronounced 'Tim toady').

**Comment by bnilsson11** [comment]
Hi,

Trying to update ArchLinux on my RP3, but it does not seem to work:

RP3DAC:~  #  sudo pacman -Syu    
:: Synchronizing package databases...
error: failed retrieving file 'RR.db' from rern.github.io : The requested URL returned error: 404
error: failed to update RR (failed to retrieve some files)
 core is up to date
 extra is up to date
 community is up to date
 alarm is up to date
 aur is up to date
error: failed to synchronize all databases

Am I doing something wrong?

BN

PS: Same in RP4 rAudio e6

**Reply by klassikmann** [reply]
Please show the result of "uname -a" and the contents of /etc/pacman.conf

**Reply by bnilsson11** [reply]
uname -a:
Linux RP3DAC 5.4.74-1-ARCH #1 SMP PREEMPT Wed Nov 4 16:43:09 UTC 2020 armv7l GNU/Linux

`/etc/pacman.conf:

#See the pacman.conf(5) manpage for option and repository directives

#GENERAL OPTIONS

[options]
#The following paths are commented out with their default values listed.
#If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
#CacheDir    = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg     = pacman glibc
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
Architecture = armv7h

#Pacman won't upgrade packages listed in IgnorePkg and members of IgnoreGroup
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

#Misc options
#UseSyslog
#Color
#TotalDownload
CheckSpace
#VerbosePkgLists

#By default, pacman accepts packages signed by keys that its local keyring
#trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

#NOTE: You must run `pacman-key --init` before first using pacman; the local
#keyring can then be populated with the keys of all official Arch Linux ARM
#packagers with `pacman-key --populate archlinuxarm`.

#REPOSITORIES
#- can be defined here or included from another file
#- pacman will search repositories in the order defined here
#- local/custom mirrors can be added here or in separate files
#- repositories listed first will take precedence when packages
#- have identical names, regardless of version number
#- URLs will have $repo replaced by the name of the current repo
#- URLs will have $arch replaced by the name of the architecture

#Repository entries are of the format:
#[repo-name]
#Server = ServerName
#Include = IncludePath

#The header [repo-name] is crucial - it must be present and
#uncommented to enable the repo.

#The testing repositories are disabled by default. To enable, uncomment the
#repo name header and Include lines. You can add preferred servers immediately
#after the header, and they will be used before the default mirrors.

[RR]
SigLevel = Optional TrustAll
Server = https://rern.github.io/$arch

[core]
Include = /etc/pacman.d/mirrorlist

[extra]
Include = /etc/pacman.d/mirrorlist

[community]
Include = /etc/pacman.d/mirrorlist

[alarm]
Include = /etc/pacman.d/mirrorlist

[aur]
Include = /etc/pacman.d/mirrorlist

#An example of a custom package repository.  See the pacman manpage for
#tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs`

**Reply by bnilsson11** [reply]
Sorry about the STRANGE formatting, I just copy/pasted from the terminal window...

**Reply by klassikmann** [reply]
@bnilsson11 
First:
> Sorry about the STRANGE formatting, I just copy/pasted from the terminal window...

I assume that paste using ALT-CTRL-v avoids the unwanted formatting here.

To your problem:
Your kernel 5.4.74-1 seems to be pretty old. Nowadays we have 5.10.xx. Therefore I think it's your first try to a system update via "pacman -Syu". And at 20210911 rern reorganized the image files. RPi3 is no longer using a 32bit-kernel but as RPi4 a 64bit one (aarch64) and he re-arranged https://rern.github.io accordingly. This could be the cause of your problem.

My proposal:
- make a backup in the GUI (system menu)
- download the latest released image ( i20211022) from https://github.com/rern/rAudio-1/releases 
- write the image to a new SDcard and boot (save the old SDcard)
- update GUI (addons menu)
- restore backup in the GUI (system menu)

Now you be capable to use "pacman -Syu".
But let @rern have a look on the topic and my advice.

**Reply by bnilsson11** [reply]
Ok, that is very good news. So the new image supports both RP3 and RP4?
I will try your advice immediately.

**Reply by klassikmann** [reply]
@bnilsson11 
> So the new image supports both RP3 and RP4?

Yes.

**Reply by rern** [reply]
@bnilsson11,

To properly format as code or file content:
- Select the content part
- Tap/click `<>` icon on the top of "Write" tab
- Verify with "Preview" tab

Please edit your post.

**Reply by bnilsson11** [reply]
Done. The '#' invoked some unexpected control codes, sorry.

**Reply by bnilsson11** [reply]
I have now updated rAudio ArchLinux without problems.
Thanks for the support.

BN
